### ISSUE TITLE
Can't open more than one JACK port

### ISSUE THREAD

**Issue by tomek-szczesny** [thestk/rtmidi#226]
Hi,

I'm getting my hands dirty trying to implement multiple MIDI port support for Giada.
I can easily enough construct the first RtMidiIn / RtMidiOut pair that I use for gathering information about available ports etc. I never open ports on them.
Any other RtMidiIn/Out instance I try to create, I get the following errors:
```
MidiInJack::initialize: JACK server not running?

MidiInJack::initialize: JACK server not running?

MidiInJack::openPort: JACK error creating port
```
As far as I'm concerned, the first line appears when I create a new RtMidiIn instance (although does not throw error for any reason).
Second and third lines appear right after calling `openPort()`, throwing exception this time.

The problem is, exactly the same code works fine with ALSA (just replaced Api where applicable). 
jackdbus.log confirms that the first port pair is created, and does not report any errors or failed attempts. 
Both of these facts raised my concern that there might be something wrong with RtMidi itself.

To complete my issue, I'm working on Ubuntu Studio 19.10, JACK of course, RtMidi 3.0.0.

I am aware that this would be nicer of me to provide a pull request instead, but I'd like to reach out first and see if this is a known issue, or might be somehow explained as my error or something.

**Comment by SpotlightKid** [comment]
Show your code.

**Comment by tomek-szczesny** [comment]
Here you go:
https://github.com/tomek-szczesny/giada/blob/e0d072b741d5acf459e0ede799114cbf2ad6ee7f/src/core/midiPorts.cpp#L266

**Comment by SpotlightKid** [comment]
Not exactly a minimal failing example. I can't even build your branch (`midi-overhaul`, I guess?). It fails at the linking stage.

You need to provide more information & guidance or it will be very hard to help you.

**Comment by tomek-szczesny** [comment]
Terribly sorry, I didn't expect you to compile this thing. Yes, I should have warned you this is work in progress.

I'll prepare a sample code later today then. Thanks for your effort.

**Comment by tomek-szczesny** [comment]
Okay, so I've spent a fair amount of time and I couldn't recreate my problem in a separate program. Finally figured it out. Port names were longer than 63 characters that Jack expects. 
So for the sake of this issue report, let's just say that these error messages were very misleading. ;)

Anyway, thanks a bunch for your attitude!

**Comment by SpotlightKid** [comment]
Yes, I know, I can be harsh sometimes. Give a man a fish and all that... ;)

**Comment by tomek-szczesny** [comment]
I completely understand that, I've searched RtMidi's issues/requests section first ;)
What I was trying to say is I'm happy that someone actually wanted to help me out around here. :)
Hopefully that was my last problem with RtMidi though :D

**Comment by radarsat1** [comment]
> Port names were longer than 63 characters that Jack expects.

Is this a check we could add in RtMidi?  Does Jack give any signal that the names are too long?

**Comment by tomek-szczesny** [comment]
I believe RtMidi could do this check and return a more meaningful error. 
I don't remember how I figured it out, but I think I saw no explicit explanation for this anywhere. Just had a bunch of controllers and one happened to have a longer name, or something.

**Comment by SpotlightKid** [comment]
> Does Jack give any signal that the names are too long?

The jack API does even have a function for that:

https://jackaudio.org/api/group__ClientFunctions.html#gab8b16ee616207532d0585d04a0bd1d60

**Comment by insolace** [comment]
The requested check is implemented at `RtMidi.cpp:4617-4620`
